utils/stats.py

```diff
@@ -0,0 +1,41 @@
+from functools import wraps
+from time import time
+
+
+class Counter:
+    def __init__(self):
+        self.__timestamp_last_cached: float = 0
```

Comment:
How do we get the stats out of this? Can we expose some kind of method to poll?

Reply:
Yes, wanted to make sure the stats are ok, will create methods to return them.